src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/LanguageServerWorkspace.cs

```diff
@@ -78,7 +85,15 @@ internal override ValueTask TryOnDocumentOpenedAsync(DocumentId documentId, Sour
         return this.ProjectSystemProjectFactory.ApplyChangeToWorkspaceAsync(
             _ =>
             {
-                this.OnDocumentOpened(documentId, textContainer, isCurrentContext, requireDocumentPresentAndClosed: false);
+                if (CurrentSolution.ContainsDocument(documentId))
+                {
+                    this.OnDocumentOpened(documentId, textContainer, isCurrentContext, requireDocumentPresentAndClosed: false);
+                }
+                else if (CurrentSolution.ContainsAdditionalDocument(documentId) && !this.IsDocumentOpen(documentId))
```

Comment:
i don't understand the IsDocumetnOpen part.

Reply:
also, i would just keep the pattern consistent.  this is def confusing.

Reply:
OK, that would require adding the requireDocumentPresentAndClosed parameter to OnAdditionalDocumentOpened as well. I didn't want to change that API so I had to do the check here

Reply:
Done